miri-script/src/commands.rs

```diff
@@ -0,0 +1,711 @@
+
+        // We configure dev builds to not be unusably slow.
+        let devel_opt_level = std::env::var("CARGO_PROFILE_DEV_OPT_LEVEL")
+            .ok()
+            .filter(|val| !val.is_empty())
+            .unwrap_or_else(|| "2".to_owned());
+        ret.extra_flags.insert("CARGO_PROFILE_DEV_OPT_LEVEL".to_owned(), devel_opt_level);
+
+        let rustflags = std::env::var("RUSTFLAGS").unwrap_or_default();
+        let rustflags = format!(
+            "-Zunstable-options -Wrustc::internal -Wrust_2018_idioms -Wunused_lifetimes -Wsemicolon_in_expressions_from_macros {rustflags}"
+        );
+        let rustflags =
+            format!("-C link-args=-Wl,-rpath,{} {rustflags}", ret.extra_vars.libdir.display());
+        ret.extra_flags.insert("RUSTFLAGS".to_owned(), rustflags);
+        ret.execute()
+    }
+    fn execute(&mut self) -> Result<()> {
+        match self.command {
+            Subcommands::Install { flags } => self.install(&flags),
+            Subcommands::Build { flags } => self.build(&flags),
+            Subcommands::Check { flags } => self.check(&flags),
+            Subcommands::Test { flags } | Subcommands::Bless { flags } => self.test(&flags),
+            Subcommands::Run { flags } | Subcommands::RunDep { flags } => self.run_(&flags),
+            Subcommands::Fmt { flags } => self.fmt(&flags),
+            Subcommands::Clippy { flags } => self.clippy(&flags),
+            Subcommands::Cargo { flags } => self.cargo(&flags),
+            Subcommands::ManySeeds { command, seed_start, seeds } =>
+                self.many_seeds(&command, *seed_start, *seeds),
+            Subcommands::Bench { benches } => self.bench(&benches),
+            Subcommands::Toolchain { flags } => self.toolchain(&flags),
+            Subcommands::RustcPull { commit } => self.rustc_pull(commit.clone()),
+            Subcommands::RustcPush { github_user, branch } =>
+                self.rustc_push(&github_user, &branch),
+        }
+    }
+    fn find_sysroot(&mut self) {
+        let current_sysroot = std::env::var("MIRI_SYSROOT").unwrap_or_default();
+
+        if !current_sysroot.is_empty() {
+            // Sysroot already set, use that.
+            let current_value =
+                self.extra_flags.insert("MIRI_SYSROOT".to_string(), current_sysroot.clone());
+            assert!(current_value.is_none() || current_value.unwrap() == current_sysroot);
+            return;
+        }
+        let miri_test_target = std::env::var("MIRI_TEST_TARGET").unwrap_or_default();
+        let target = (!miri_test_target.is_empty()).then_some(miri_test_target.as_str());
+        let sysroot = self.build_sysroot(target);
+        self.extra_flags.insert("MIRI_SYSROOT".into(), sysroot);
+    }
+
+    fn build_sysroot(&self, target: Option<&str>) -> String {
+        let manifest_path = path!(self.miri_dir / "cargo-miri" / "Cargo.toml");
+        let manifest_path = manifest_path.to_string_lossy().to_string();
```

Comment:
```suggestion
        let manifest_path = manifest_path.to_string_lossy().into_owned();
```
applies to all `Cow` (`to_string` goes through `Display` so this is extremely generic and not usually what should be used IMO)